lib/chef/knife/ec2_server_create.rb

```diff
@@ -524,42 +509,42 @@ def validation_key_tmpfile
 
       def download_validation_key(tempfile)
         Chef::Log.debug "Downloading validation key " \
-          "<#{Chef::Config[:knife][:validation_key_url]}> to file " \
+          "<#{config[:validation_key_url]}> to file " \
           "<#{tempfile}>"
 
-        case URI(Chef::Config[:knife][:validation_key_url]).scheme
+        case URI(config[:validation_key_url]).scheme
         when "s3"
           File.open(tempfile, "w") { |f| f.write(s3_validation_key) }
         end
       end
 
       def s3_validation_key
         @s3_validation_key ||= begin
-          Chef::Knife::S3Source.fetch(Chef::Config[:knife][:validation_key_url])
+          Chef::Knife::S3Source.fetch(config[:validation_key_url])
         end
       end
 
       def s3_secret
         @s3_secret ||= begin
-          return false unless config_value(:s3_secret)
+          return false unless config[:s3_secret]
 
-          Chef::Knife::S3Source.fetch(config_value(:s3_secret))
+          Chef::Knife::S3Source.fetch(config[:s3_secret])
         end
       end
 
       def bootstrap_common_params
-        config[:encrypted_data_bag_secret] = s3_secret || config_value(:secret)
-        config[:encrypted_data_bag_secret_file] = config_value(:secret_file)
+        config[:encrypted_data_bag_secret] = s3_secret || config[:secret]
+        config[:encrypted_data_bag_secret_file] = config[:secret_file]
         # retrieving the secret from S3 is unique to knife-ec2, so we need to set "command line secret" to the value fetched from S3
         # When linux vm is spawned, the chef's secret option proc function sets the value "command line secret" and this value is used by
```

Comment:
Is this comment still valid?

Reply:
yeah i think it actually still is.

there's still the distinction between `config[:cl_secret]` (set by the command line) and `config[:secret]` (the approved API for the Chef::Config option):

https://github.com/chef/chef/blob/d657a69f13d97f2ae89f39b6632f576b13822c7d/lib/chef/knife/data_bag_secret_options.rb#L65-L75

i went through that a few weeks back and convinced myself i couldn't break that for some reason, but i've blotted it out of my mind by now.